README.md

```diff
@@ -1,2 +1,159 @@
+| keep time, pageviews, url
+| sort pageviews -desc
+| head 10
+| view barchart;
+```
+
+### Show the pages most viewed by new users by referral source over the past week
+
+```juttle
+read ga -last :w: -viewId 78763287
+| reduce sum(pageviews) by userType, pagePath, source
+| filter source != '(direct)' AND userType='New Visitor'
+| sort sum -desc
+| head 10
+```
+
+### List all the web properties and views in the account
+
+```juttle
+read ga | reduce by webProperty, webPropertyId, view, viewId
+```
+
+### Print the user counts for each view for a given site
+
+```juttle
+read ga -webProperty 'www.mysite.io'
+| reduce users=sum(users) by view, viewId
+```
+
+## Installation
+
+Like Juttle itself, the adapter is installed as a npm package. Both Juttle and the adapter need to be installed side-by-side:
+
+```bash
+$ npm install juttle
+$ npm install juttle-ga-adapter
+```
+
+Alternatively you could install the juttle-engine which includes the adapter:
+
+```bash
+$ npm install juttle-engine
+```
+
+## Authorization / Configuration
+
+First you need to enable the API and create a service account for authentication.
+
+The process is enumerated in step 1 of the [Hello Analytics API Guide](https://developers.google.com/analytics/devguides/reporting/core/v3/quickstart/service-java#summary_auth). You don't need to do steps 2-4 in that guide.
+
+Once you've created the account and are ready to download the credentials, make sure you select the `JSON` key type and not the `P12` key type. This should result in a file downloaded to your computer.
+
+Next the adapter needs to be registered and configured so that it can be used from within Juttle.
+
+To do so, take the contents of the service account JSON file and add it to a new "ga" section in the adapters configuration of your `~/.juttle/config.json` file as follows:
+
+```json
+{
+    "adapters": {
+        "ga": {
+            "service_account": {
+                "type": "service_account",
+                "project_id": "<YOUR PROJECT ID>",
+                "private_key_id": "<YOUR PROJECT KEY ID>",
+                "private_key": "-----BEGIN PRIVATE KEY-----\n<YOUR PRIVATE KEY>\n-----END PRIVATE KEY-----\n",
+                "client_email": "<YOUR ACCOUNT EMAIL>",
+                "client_id": "<YOUR CLIENT ID>",
+                "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+                "token_uri": "https://accounts.google.com/o/oauth2/token",
+                "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+                "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/<YOUR ACCOUNT EMAIL>"
+            }
+
+        }
+    }
+}
+```
+
+## Usage
+
+### Read options
+
+Since google analytics returns aggregate data, the only supported query pattern using Juttle uses a combination of [read](http://juttle.github.io/juttle/processors/read/) and [reduce](http://juttle.github.io/juttle/processors/reduce/), using the general form:
```

Comment:
This was a bit vague for me. I think you're saying that ga only provides summary data and not raw data, making a subsequent reduce the best way to map the summary back into the read?